test/pkg/node/node.go

```diff
@@ -0,0 +1,118 @@
+package node
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/onsi/gomega"
+
+	"github.com/openshift-kni/commatrix/pkg/client"
+	"github.com/openshift-kni/commatrix/pkg/consts"
+	"github.com/openshift-kni/commatrix/pkg/utils"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type NodeWrapper struct {
+	Node *v1.Node
+	cs   *client.ClientSet
+}
+
+func New(node *v1.Node, cs *client.ClientSet) *NodeWrapper {
+	return &NodeWrapper{
+		Node: node,
+		cs:   cs,
+	}
+}
+
+// SoftRebootNodeAndWaitForDisconnect soft reboots given node and wait for node to be unreachable.
+func (node *NodeWrapper) SoftRebootNodeAndWaitForDisconnect(ns string) error {
+	utilsHelpers := utils.New(node.cs)
+	err := utilsHelpers.CreateNamespace(ns)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		if err := utilsHelpers.DeleteNamespace(ns); err != nil {
+			fmt.Printf("failed cleaning namespace %s: %v", ns, err)
+		}
+	}()
+
+	debugPod, err := utilsHelpers.CreatePodOnNode(node.Node.Name, ns, consts.DefaultDebugPodImage)
+	if err != nil {
+		return fmt.Errorf("failed to create debug pod on node %s: %w", node.Node.Name, err)
+	}
+
+	defer func() {
+		err := utilsHelpers.DeletePod(debugPod)
+		if err != nil {
+			fmt.Printf("failed cleaning debug pod %s: %v", debugPod, err)
+		}
+	}()
+
+	rebootcmd := []string{"chroot", "/host", "reboot"}
+
+	out, err := utilsHelpers.RunCommandOnPod(debugPod, rebootcmd)
+	if err != nil {
+		return fmt.Errorf("failed to install nftables on  debugpod on node%s: %w", node.Node.Name, err)
+	}
+	log.Printf("out %s is out\n", out)
+
+	node.WaitForNodeNotReady()
+	return nil
+}
+
+// WaitForNodeNotReady waits for the node to be in the NotReady state in Kubernetes.
+func (node *NodeWrapper) WaitForNodeNotReady() {
+	log.Printf("Waiting for node %s to be in NotReady state", node.Node.Name)
+
+	timeout := 5 * time.Minute
+	gomega.Eventually(func() bool {
+		updatedNode, err := node.cs.Nodes().Get(context.TODO(), node.Node.Name, metav1.GetOptions{})
```

Comment:
Just an example of why I think this struct is not needed and not used, especially that this is an internal func. This is very confusing and makes the code less readable due to the use of the struct, instead of simply having:
cs.Nodes().Get(context.TODO(), node.Name, metav1.GetOptions{})